pkgs/development/compilers/ghc/8.10.4.nix

```diff
@@ -38,13 +38,31 @@
   ghcFlavour ? lib.optionalString (stdenv.targetPlatform != stdenv.hostPlatform)
     (if useLLVM then "perf-cross" else "perf-cross-ncg")
 
+, #  Whether to build sphinx documentation.
+  enableDocs ? (
+    # Docs disabled for musl and cross because it's a large task to keep
+    # all `sphinx` dependencies building in those environments.
+    # `sphinx` pullls in among others:
+    # Ruby, Python, Perl, Rust, OpenGL, Xorg, gtk, LLVM.
```

Comment:
> OpenGL, Xorg, gtk, LLVM

(FYI) This seems to be mostly because of `librsvg`. Issue https://github.com/NixOS/nixpkgs/issues/126611

Regardless, I don't think we want the other dependencies for cross, so +1

Reply:
Sphinx is only used at build time, so all these should be no problem except for annoying bugs where stuff is target sensative.

Reply:
@Ericson2314 Yes, the problem with it is that a lot of those build-time dependencies often simply have broken builds in `pkgsMusl`.

Reply:
`pkgsMusl.buildPackages` uses `glibc` though, so these *should* just work as normally.

Reply:
Sphinx definitely pulls in musl-based builds of the big stuff when put into `nativeBuildInputs`:

`nix-store -q --tree $(NIX_PATH=nixpkgs=. nix-instantiate -A pkgsMusl.haskell.compiler.ghc8104)`

```
+---/nix/store/rwknqkhd87phxq8iw1hz2kh983za44xi-python3.8-sphinx-4.0.2.drv
    +---/nix/store/ck02h2ph9zsscw3cb59brn669hbzynfm-python3.8-sphinxcontrib-serializinghtml-1.1.5.drv
    |   +---/nix/store/bl58hg9v5ni17wl5255n6gn8ryqn0ncr-librsvg-2.50.6.drv
    |   |   +---/nix/store/dqb7wllys8fi4c05kayp7vis7q7sqyv5-rustc-1.52.1.drv
```

`nix-store -q --tree $(NIX_PATH=nixpkgs=. nix-instantiate -A haskell.compiler.ghc8104)`

```
        +---/nix/store/pr7m6hbnn9xhcwmq1mnxcygnm06qkfgk-librsvg-2.50.6.drv
```

Diffing those two librsvg derivations:

`nix-diff /nix/store/pr7m6hbnn9xhcwmq1mnxcygnm06qkfgk-librsvg-2.50.6.drv /nix/store/bl58hg9v5ni17wl5255n6gn8ryqn0ncr-librsvg-2.50.6.drv`

```
        • The input named `bootstrap-stage0-binutils-wrapper-` differs
          - /nix/store/pmvn7qs31f8fwl04w6qdrzb6kqsjizbg-bootstrap-stage0-binutils-wrapper-.drv:{out}
          + /nix/store/fz2w0rj4rvmbny1hrz8hc0hlhjahffrs-bootstrap-stage0-binutils-wrapper-.drv:{out}
          • The builders do not match
              - /nix/store/p4s4jf7aq6v6z9iazll1aiqwb34aqxq9-bootstrap-tools/bin/bash
              + /nix/store/5b7a4ynm23gm375q1fzbsh4sqc806864-bootstrap-tools/bin/bash
          • The set of input names do not match:
              - bootstrap-stage0-glibc-bootstrap
              + bootstrap-stage0-musl-bootstrap
```

`glibc-bootstrap` vs `musl-bootstrap` in here.

Reply:
Resolved on Matrix: https://matrix.to/#/!RbXGJhHMsnQcNIDFWN:nixos.org/$2dtglPPokoMcxS7yQt53iBLFvMkqr-R3yqdYA5hJyVg?via=nixos.org&via=matrix.org&via=orbstheorem.ch

> oh yeah it isn't cross compiled, nevermind
>
> https://github.com/NixOS/nixpkgs/blob/6cc55fd6380dd892f1d41ba7d07efa2a813d38fe/pkgs/top-level/stage.nix#L178-L193